rt/transports/jms/src/main/java/org/apache/cxf/transport/jms/util/PollingMessageListenerContainer.java

```diff
@@ -210,6 +210,8 @@ protected void handleException(Throwable e) {
             wrapped.addSuppressed(e);
         }
         this.exceptionListener.onException(wrapped);
+        // stop the executor and the current thread after handling of the exception.
```

Comment:
If I understood the idea behind current implementation correctly, we have the problem with `running` flag. The first thing `restartConnection` does is to deactivate the existing `jmsListener` by calling shutdown. Unfortunately, method short-circuits because the `running` flag is unset, the executor is not being shut down properly. We could fix it by introducing `shutdown` flag or just calling `super.stop();`:
```
@Override
    public void stop() {
        LOG.fine("Shuttting down " + this.getClass().getSimpleName());
        if (!running) {
            super.stop();
            return;
        }
        running = false;
        super.stop();
    }
```

Wdyt?

Reply:
Yes. I had two reasons for my solution:
* If you call `stop()` before the Exception handling is finished, the poller thread will be shutdown and the Exception handling will be interrupted immediately. (`AbstractMessageListenerContainer#stop()` calls `shutdownNow()`)
* `JMSDestination#restartConnection()` checks the running-Flag now, so we need to unset it before `ExceptionListener` calls `restartConnection()`

If it's ok for you I change it to 
   ```
 @Override
    public void stop() {
        LOG.fine("Shutting down " + this.getClass().getSimpleName());
        running = false;
        super.stop();
    }
```
and to `this.stop()` without comment in `handleException`?

Reply:
:+1: , should be a bit simpler overall I think, thanks!

Reply:
done